### ISSUE TITLE
test failure due to data race

### ISSUE THREAD

**Issue by olix0r** [runconduit/conduit#278]
```
==================
WARNING: DATA RACE
Read at 0x00c4201f8218 by goroutine 36:
  github.com/runconduit/conduit/controller/api/public.(*mockGrpcServer).Tap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:53 +0x115
  github.com/runconduit/conduit/controller/api/public.(*handler).handleTap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:158 +0x18b
  github.com/runconduit/conduit/controller/api/public.(*handler).ServeHTTP()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:50 +0x4db
  net/http.serverHandler.ServeHTTP()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2619 +0xbc
  net/http.(*conn).serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1801 +0x83b
Previous write at 0x00c4201f8218 by goroutine 48:
  github.com/runconduit/conduit/controller/api/public.TestServer.func4()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:179 +0xcd
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
Goroutine 36 (running) created at:
  net/http.(*Server).Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2720 +0x37c
  net/http.Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2323 +0xe2
  github.com/runconduit/conduit/controller/api/public.TestServer.func1()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:75 +0x5c
Goroutine 48 (finished) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:789 +0x568
  github.com/runconduit/conduit/controller/api/public.TestServer()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:178 +0x58a
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Read at 0x00c420466de0 by goroutine 36:
  errors.(*errorString).Error()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/errors/errors.go:19 +0x3b
  github.com/runconduit/conduit/controller/api/public.writeErrorToHttpResponse()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/proto_over_http.go:71 +0x163
  github.com/runconduit/conduit/controller/api/public.(*handler).handleTap()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:160 +0x1c2
  github.com/runconduit/conduit/controller/api/public.(*handler).ServeHTTP()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server.go:50 +0x4db
  net/http.serverHandler.ServeHTTP()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2619 +0xbc
  net/http.(*conn).serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1801 +0x83b
Previous write at 0x00c420466de0 by goroutine 48:
  github.com/runconduit/conduit/controller/api/public.TestServer.func4()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/client.go:5 +0x93
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
Goroutine 36 (running) created at:
  net/http.(*Server).Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2720 +0x37c
  net/http.Serve()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2323 +0xe2
  github.com/runconduit/conduit/controller/api/public.TestServer.func1()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:75 +0x5c
Goroutine 48 (finished) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:789 +0x568
  github.com/runconduit/conduit/controller/api/public.TestServer()
      /home/travis/gopath/src/github.com/runconduit/conduit/controller/api/public/http_server_test.go:178 +0x58a
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.linux.amd64/src/testing/testing.go:746 +0x16c
==================
=== RUN   TestDeserializePayloadFromReader/Returns_error_when_message_has_fewer_bytes_than_declared_message_size
--- FAIL: TestDeserializePayloadFromReader (0.05s)
    --- PASS: TestDeserializePayloadFromReader/Can_read_message_correctly_based_on_payload_size_correct_payload_size_to_message (0.00s)
    --- PASS: TestDeserializePayloadFromReader/Can_multiple_messages_in_the_same_stream (0.03s)
    --- PASS: TestDeserializePayloadFromReader/Can_write_and_read__marshalled_protobuf_messages (0.01s)
    --- PASS: TestDeserializePayloadFromReader/Can_read_byte_streams_larger_than_Go's_default_buffer_chunk_size (0.02s)
    --- PASS: TestDeserializePayloadFromReader/Returns_error_when_message_has_fewer_bytes_than_declared_message_size (0.00s)
	testing.go:699: race detected during execution of test
```

**Comment by pcalcado** [comment]
https://github.com/runconduit/conduit/pull/281/commits/8fa9386b53fdc447cb09aceeb2cc70ab83761c77 seems to have resolved some of the http issues. I am thinking that there is some difference between how error is impleented in linux vs mac, still investigating

**Comment by pcalcado** [comment]
Closed by #282
